src/main/java/com/google/sps/data/WaterSystem.java

```diff
@@ -126,7 +127,7 @@ public WaterSystem(Entity entity) throws EntityNotFoundException, IOException {
         this.county = (String) entity.getProperty(COUNTY_PROPERTY);
         this.populationServed = ((Long) entity.getProperty(POPULATION_PROPERTY)).intValue();
         this.contaminants = new ArrayList<WaterContaminant>();
-        byte[] contaminantsData = Base64.getDecoder().decode((String)entity.getProperty(CONTAMINANTS_PROPERTY));
+        byte[] contaminantsData = Base64.getDecoder().decode(((Text)entity.getProperty(CONTAMINANTS_PROPERTY)).getValue());
```

Comment:
nit: You're on a good track describing the high level changes in your PR and commit meesages. One way to improve the quality of these messages though is to explain why the changes are being made. Here, you can expand on why you're switching from String to Text.

Given this project is close to being complete you don't really have to do this here, but it's good to know for production projects.

Reply:
Understood, in this case I'm switching from String to Text because String properties have a size limit whereas Text properties do not